File: original artifact/Ian_Foss_cs-360_project_three_inventory_app/cs360_project_three/app/src/main/java/com/example/cs_360_project_three/InventoryDatabase.java
package com.example.cs_360_project_three;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;


// Database singleton class with entities for the users and inventory items
@Database(entities = {UserEntity.class, InventoryItemEntity.class}, version = 2)
public abstract class InventoryDatabase extends RoomDatabase {
    public abstract UserDao userDao();
    public abstract InventoryItemDao itemDao();

    private static volatile InventoryDatabase INSTANCE;

    public static InventoryDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            synchronized (InventoryDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            InventoryDatabase.class,
                            "inventory.db")
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return INSTANCE;
    }

}

File: enhancements/enhancement-one/Enhancement One/app/src/main/java/com/example/CS499_Capstone/InventoryDatabase.kt
package com.example.CS499_Capstone

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import androidx.room.RoomDatabase.Builder
import kotlin.concurrent.Volatile

// Database singleton class with entities for the users and inventory items
@Database(entities = [UserEntity::class, InventoryItemEntity::class], version = 2)
abstract class InventoryDatabase : RoomDatabase() {
    abstract fun userDao(): UserDao
    abstract fun itemDao(): InventoryItemDao

    companion object {
        @Volatile
        private var INSTANCE: InventoryDatabase? = null

        fun getInstance(context: Context): InventoryDatabase {

            return INSTANCE ?: synchronized(this) {
                val instance = databaseBuilder(
                    context.applicationContext,
                    InventoryDatabase::class.java,
                    "inventory.db"
                )
                    .fallbackToDestructiveMigration(false)
                    .build()

                INSTANCE = instance
                instance
            }

        }
    }
}
